zone/groups.cpp

```diff
@@ -554,11 +553,10 @@ void Group::MemberZoned(Mob* removemob) {
 	if(removemob == GetLeader())
 		SetLeader(nullptr);
 
-	for (i = 0; i < MAX_GROUP_MEMBERS; i++) {
-		if (members[i] == removemob) {
-			members[i] = nullptr;
-			//should NOT clear the name, it is used for world communication.
-			break;
+	//should NOT clear the name, it is used for world communication.
+	for (auto & m : members) {
+		if (m && (m == removemob || m->IsBot() && m->CastToBot()->GetBotOwner() == removemob)) {
+			m = nullptr;
 		}
```

Comment:
Add a `break;` here?

Reply:
Can't break cause we need to set any Bots that removemob is the owner of to nullptr here, or we crash later on.